app/src/main/java/com/murrayc/galaxyzoo/app/SubjectFragment.java

```diff
@@ -427,7 +438,63 @@ private void updateShareActionIntent() {
         //shareIntent.putExtra(Intent.EXTRA_STREAM, mUriImageStandard);
         //shareIntent.setType("image/*");
 
+        if (mUriStandardRemote!=null) {
+            GetImageBitmapAsyncTask getImageBitmapAsyncTask = new GetImageBitmapAsyncTask(){
+                @Override
+                protected void onPostExecute(Uri uri) {
+                    shareIntent.setType("image/*");
+                    shareIntent.putExtra(Intent.EXTRA_STREAM, uri);
+                    shareIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
+                    mShareActionProvider.setShareIntent(shareIntent);
+                }
+            };
+            getImageBitmapAsyncTask.execute(mUriStandardRemote);
+        }
+
 
         mShareActionProvider.setShareIntent(shareIntent);
     }
+
+    private static final int REQUEST_EXTERNAL_STORAGE = 1;
+    private static String[] PERMISSIONS_STORAGE = {
```

Comment:
Can you use the constants here instead of hard-coding these strings? This seems to be suitable: https://developer.android.com/reference/android/Manifest.permission.html#WRITE_EXTERNAL_STORAGE